src/app/Library/Components/Fields/TextField.php

```diff
@@ -0,0 +1,20 @@
+<?php
```

Comment:
**Question for future us.**

Would it be easier to define a custom field like this?

```php
class GutenbergField extends BaseField
{
    public static function getAttributes(): array
    {
        return [
            FieldType::class,
            LabelAttribute::class,
            HintAttribute::class,
            PrefixAttribute::class,
            SuffixAttribute::class,
        ];
    }

    public static function getAttributeValidationRules() : array
    {
        return [
            'characterCount' => 'required|boolean',
        ];
    }

    public static function getAttributeDefaults() : array
    {
        return [
            'characterCount' => true,
        ];
    }

    public static function getBlockedAttributes(): array
    {
        return ['type'];
    }

    public static function getPath()
    {
        return 'fields/gutenberg';
    }
}
``` 

or like this?

```php
class GutenbergField extends BaseField
{
    use Attributes\FieldType;
    use Attributes\LabelAttribute;
    use Attributes\HintAttribute;
    use Attributes\PrefixAttribute;
    use Attributes\SuffixAttribute;
    use Attributes\EntityCount;
    use Attributes\LabelAttribute { label as originalLabel; };

    public function label() : SmartAttribute {
        return $this->originalLabel->default(true);
    }

    public function characterCount() : SmartAttribute {
        return SmartAttribute::make('characterCount')
                            ->default(false)
                            ->value(false)
                            ->rules('required|boolean');
    }

    public static function getPath()
    {
        return 'fields/gutenberg';
    }
}
```

Reply:
PROs and CONs of the second option.

PROs:
- easier to type for the developer ;
- easier to read/understand for the developer (all code is in one place);
- automatic IDE auto-completion (because each attribute has a method);

CONs:
- it's more difficult to globally override an attribute (eg. Label for all fields);
- we might have problems if one attribute needs another attribute to be already defined, because we don't have control over the order, other than the order the traits are written in; this might result in it being impossible to create attributes dependant on other attributes;